routing/result_interpretation_test.go

```diff
@@ -45,18 +45,44 @@ var (
 	}
 )
 
+// TestResultInterpretationSuccess asserts that a successful payment result is
+// interpreted correctly.
+func TestResultInterpretationSuccess(t *testing.T) {
```

Comment:
We should also add a test case (if there isn't one already) that ensures pairs are marked as successful up to the error source for a failed payment.

Reply:
Existing fail test extended